src/game/client/components/scoreboard.cpp

```diff
@@ -103,50 +142,117 @@ void CScoreboard::RenderSpectators(float x, float y, float w)
+	float PingOffset = ScoreOffset+ScoreLength, PingLength = 35.0f;
+	float tw = 0.0f;
+
+	bool NoTitle = pTitle? false : true;
+
+	// count players
+	int NumPlayers = 0;
+	for(int i = 0; i < MAX_CLIENTS; i++)
+		if(m_pClient->m_aClients[i].m_Active && m_pClient->m_aClients[i].m_Team == Team)
+			NumPlayers++;
+	m_PlayerLines = max(m_PlayerLines, NumPlayers);
+
+	char aBuf[128] = {0};
 
 	// background
 	Graphics()->BlendNormal();
-	RenderTools()->DrawRoundRect(&Rect, vec4(0.0f, 0.0f, 0.0f, 0.5f), 17.0f);
+	vec4 Color;
+	if(Team == TEAM_RED && m_pClient->m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS)
+		Color = vec4(0.975f, 0.17f, 0.17f, 0.75f);
+	else if(Team == TEAM_BLUE)
+		Color = vec4(0.17f, 0.46f, 0.975f, 0.75f);
+	else
+		Color = vec4(0.0f, 0.0f, 0.0f, 0.5f);
+	CUIRect Rect = {x, y, w, HeadlineHeight};
+	RenderTools()->DrawRoundRect(&Rect, Color, 5.0f);
 
 	// render title
-	float TitleFontsize = 40.0f;
-	if(!pTitle)
+	if(NoTitle)
 	{
 		if(m_pClient->m_Snap.m_pGameData->m_GameStateFlags&GAMESTATEFLAG_GAMEOVER)
 			pTitle = Localize("Game over");
 		else if(m_pClient->m_Snap.m_pGameData->m_GameStateFlags&GAMESTATEFLAG_ROUNDOVER)
 			pTitle = Localize("Round over");
 		else
-			pTitle = Localize("Score board");
+			pTitle = Localize("Scoreboard");
+	}
+	else
+	{
+		if(Team == TEAM_BLUE)
+			str_format(aBuf, sizeof(aBuf), "(%d) %s", NumPlayers, pTitle);
+		else
+			str_format(aBuf, sizeof(aBuf), "%s (%d)", pTitle, NumPlayers);
+	}
+
+	if(Align == -1)
+	{
+		tw = TextRender()->TextWidth(0, TitleFontsize, pTitle, -1);
+		TextRender()->Text(0, x+20.0f, y+5.0f, TitleFontsize, pTitle, -1);
+		if(!NoTitle)
+		{
+			str_format(aBuf, sizeof(aBuf), " (%d)", NumPlayers);
+			TextRender()->TextColor(1.0f, 1.0f, 1.0f, 0.5f);
+			TextRender()->Text(0, x+20.0f+tw, y+5.0f, TitleFontsize, aBuf, -1);
+			TextRender()->TextColor(1.0f, 1.0f, 1.0f, 1.0f);
+		}
+	}
+	else
+	{
+		tw = TextRender()->TextWidth(0, TitleFontsize, pTitle, -1);
+		if(!NoTitle)
+		{
+			str_format(aBuf, sizeof(aBuf), "(%d) ", NumPlayers);
+			float PlayersTextWidth = TextRender()->TextWidth(0, TitleFontsize, aBuf, -1);
+			TextRender()->TextColor(1.0f, 1.0f, 1.0f, 0.5f);
+			TextRender()->Text(0, x+w-tw-PlayersTextWidth-20.0f, y+5.0f, TitleFontsize, aBuf, -1);
+			TextRender()->TextColor(1.0f, 1.0f, 1.0f, 1.0f);
+		}
+		tw = TextRender()->TextWidth(0, TitleFontsize, pTitle, -1);
+		TextRender()->Text(0, x+w-tw-20.0f, y+5.0f, TitleFontsize, pTitle, -1);
 	}
-	TextRender()->Text(0, x+20.0f, y, TitleFontsize, pTitle, -1);
 
-	char aBuf[128] = {0};
 	if(m_pClient->m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS)
 	{
 		int Score = Team == TEAM_RED ? m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreRed : m_pClient->m_Snap.m_pGameDataTeam->m_TeamscoreBlue;
 		str_format(aBuf, sizeof(aBuf), "%d", Score);
 	}
 	else
 	{
-		if(m_pClient->m_Snap.m_SpecInfo.m_Active && m_pClient->m_Snap.m_SpecInfo.m_SpectatorID >= 0 && m_pClient->m_Snap.m_SpecInfo.m_SpectatorID < MAX_CLIENTS &&
```

Comment:
This line was correct. 'SPEC_FREEVIEW' is 0 and used for SpecModes.

Reply:
Changed. I keep the `m_pClient->m_Snap.m_paPlayerInfos[m_pClient->m_Snap.m_SpecInfo.m_SpectatorID]` check right?